src/Draco.Compiler/Syntax/Parser.cs

```diff
@@ -32,5 +34,104 @@ public CompilationUnit ParseCompilationUnit()
         return new(decls.ToValue());
     }
 
-    private Decl ParseDeclaration() => throw new NotImplementedException();
+    private Decl ParseDeclaration()
+    {
+        var keyword = this.tokenSource.Peek();
+        if (keyword.Type == TokenType.KeywordFunc)
+        {
+            return this.ParseFunction();
+        }
+        else if (keyword.Type == TokenType.KeywordVal)
```

Comment:
If the function parsing is in its dedicated function, why is variable declaration inlined? I'd have a `ParseVariableDeclaration()` that handles both keywords.